File: backend/components/metiers/threed/plugins/samples/sample_plugin.py
"""
sample_plugin.py
Exemple de plugin (sample) pour le module core plugins threed (Python).
Sert de référence pour l'intégration, la documentation et les tests.
"""

class SamplePlugin:
    def __init__(self, options=None):
        self.options = options or {}
        self.config = None

    def init(self, config):
        """Initialise le sample plugin avec des paramètres de test."""
        self.config = config
        # Logique d'initialisation de sample
        return True

    def run(self, data):
        """Exécute la fonctionnalité principale du sample plugin."""
        # Traitement métier de sample
        return {"processed": True, "data": data, "config": self.config}

File: backend/components/metiers/administration_publique/plugins/samples/sample_plugin.js
// sample_plugin.js
// Exemple de plugin (sample) pour le module core plugins threed.
// Sert de référence pour l'intégration, la documentation et les tests.

class SamplePlugin {
  constructor(options = {}) {
    this.options = options;
  }

  /**
   * Initialise le sample plugin avec des paramètres de test.
   * @param {Object} config - Configuration de test.
   */
  init(config) {
    this.config = config;
    // Logique d'initialisation de sample
    return true;
  }

  /**
   * Exécute la fonctionnalité principale du sample plugin.
   * @param {any} data - Données à traiter.
   * @returns {any} Résultat du traitement sample.
   */
  run(data) {
    // Traitement métier de sample
    return { processed: true, data, config: this.config };
  }
}

module.exports = SamplePlugin;
